### PR TITLE
ENH: Add Python3.10

### PR DESCRIPTION
**Pull request by larsoner**
Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* [ ] ~~Reset the build number to `0` (if the version changed)~~
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

### CODE REVIEW

**Comment by larsoner** [comment]
@conda-forge-admin, please rerender

**Comment by larsoner** [comment]
@conda-forge-admin add python 3.10

**Comment by hoechenberger** [comment]
Re-rendering won't suffice, you'll need to add the migration file:

https://github.com/regro-cf-autotick-bot/dipy-feedstock/blob/ea7d8214e9d0e3be300fa4016018f1ced6ee06d5/.ci_support/migrations/python310.yaml

**Comment by larsoner** [comment]
@hoechenberger can you push commits / comments to do it if you know how?

**Comment by hoechenberger** [comment]
@larsoner I cannot push, but the conda-forge folks told me on gitter the we could simply add the "migration file" manually and then re-render and it should be good.

I think the file they were referring to this:
https://raw.githubusercontent.com/regro-cf-autotick-bot/dipy-feedstock/ea7d8214e9d0e3be300fa4016018f1ced6ee06d5/.ci_support/migrations/python310.yaml

It should be added as `.ci_support/migrations/python310.yaml`

**Comment by hoechenberger** [comment]
> I think the file they were referring to this:

Yep, just got confirmation.

**Comment by hoechenberger** [comment]
@conda-forge-admin, please rerender

**Comment by larsoner** [comment]
@conda-forge-admin, please rerender

File: .ci_support/linux_64_numpy1.21python3.10.____cpython.yaml [hunk 1]
```diff
@@ -0,0 +1,25 @@
+c_compiler:
```
Comment (larsoner) [inline .ci_support/linux_64_numpy1.21python3.10.____cpython.yaml:1]: Hooray!

**Comment by hoechenberger** [comment]
Yesss, it seems to be working, @larsoner!!

**Comment by hoechenberger** [comment]
@larsoner All green!

File: .ci_support/migrations/python310.yaml [hunk 2]
```diff
@@ -0,0 +1,34 @@
+migrator_ts: 1634137107
```
Comment (skoudoro) [inline .ci_support/migrations/python310.yaml:1]: I think you should remove this migration file, it should not be included.

**Review by skoudoro — APPROVED** [review]
Thank you a lot @larsoner and @hoechenberger for this!

I do not think that the migration files should be included, but I will go ahead and merge this PR.

I do not want you to wait more and this can be deleted later.

Thanks!

**Pull request merged**